### PR TITLE
Create changelog in documentation

### PR DESCRIPTION
**Pull request by rudigiesler**
We need to add a changelog to the documentation to document what changed with each release of Junebug.

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
+.. _release-notes
+
+Release Notes
+=============
+
+v1.0.0
```
Comment (rudigiesler) [inline docs/release-notes.rst]: Should this be `v1.0.0`, or `v0.1.0`?

**Comment by rudigiesler** [comment]
Ready for review.

File: docs/release-notes.rst (hunk 1 above)
Comment (hodgestar) [inline docs/release-notes.rst]: I think we're planning to do a splash `v1.0.0` release at some point in the future, so I'd vote for `v0.x.0` for some value of `x` less than 7 or so.

**Comment by hodgestar** [comment]
Other than the version question, looks good.

File: docs/release-notes.rst (hunk 1 above)
Comment (rudigiesler) [inline docs/release-notes.rst]: Okay I'll do the `0.1.0` version then. I wanted something to signify that we've finished all the work that we originally set out to do, and not just continue with `0.0.6`.

**Comment by rudigiesler** [comment]
Version changed, ready for review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**